opencog/atoms/base/Link.cc

```diff
@@ -159,6 +159,16 @@ bool Link::operator<(const Atom& other) const
     return false;
 }
 
+bool Link::operator<(const Value& other) const
+{
+	if (get_type() != other.get_type())
+		return get_type() < other.get_type();
+
+	const Link* lov = (const Link*) &other;
+
+	return this < lov;
```

Comment:
It was my intention that this would call operator<(const Atom& other) but now that you bring it up there should probably be a test for it or something since it could validly call operator<(const Value& other) which would be an infinite loop.

The operator<(const Atom& other) uses a hashbased comparison so I thought that comment makes sense but I guess I can just say that it uses operator<(const Atom& other).

Reply:
I understand but `Link::operator<(const Value&)` as implemented here does not call `Link::operator<(const Atom&)` instead it just compares their addresses.

Reply:
Yeah, I really don't want this. We used to do address compares exactly like this, some umpteen decades ago, and it was found wanting, troublesome, prone to bugs and mis-use. (I spent nearly a year trying to brow-beat Nil int thinking that address-compare was a good idea, until I threw in the towel.)  The atom-hash was invented in order to solve this problem. Please use that.

Reply:
oh, hang on, my last comment isn't right. Deleting it.